compiler/src/dotty/tools/dotc/tastyreflect/TreeOpsImpl.scala

```diff
@@ -776,7 +776,7 @@ trait TreeOpsImpl extends scala.tasty.reflect.TreeOps with CoreImpl with Helpers
         tpd.Return(expr, ctx.owner)
 
       def copy(original: Tree)(expr: Term)(implicit ctx: Context): Return =
-        tpd.cpy.Return(original)(expr, tpd.EmptyTree)
+        tpd.cpy.Return(original)(expr, tpd.ref(ctx.owner))
```

Comment:
Ahead warning: We'll probably drop non-local returns so this will change at some point.

Reply:
I am aware of this, that is why it always references the owner. If we do not drop non-local returns I would have to add it as an extra parameter to `copy`, `apply` and `unapply`.